proposals/1984_ca-bundle-from-secret-ref/README.md

```diff
@@ -11,124 +11,261 @@ Once approved and implemented, they become historical documents.
+This function will continue to set the `TLSConfigurationValid` condition based on the validity.
 
 ### Concierge
+For Pinniped Concierge, the `TLSSpec` is generated using the [apis/concierge/authentication/v1alpha1/types_tls.go.tmpl](https://github.com/vmware-tanzu/pinniped/blob/main/apis/concierge/authentication/v1alpha1/types_tls.go.tmpl). This template will be updated to allow Pinniped Supervisor custom resources to source certificate authority data from kubernetes secrets or configmaps.
 
-* `CredentialIssuer.spec.impersonationProxy.tls` is used to serve TLS.
-  It can be ignored for this proposal, since it already uses an external secret.
-* `CredentialIssuer.status.kubeConfigInfo.certificateAuthorityData` is a deprecated status output.
-  It should be either ignored or removed based on this proposal.
-* `CredentialIssuer.status.strategies[*].frontend.tokenCredentialRequestInfo.certificateAuthorityData` is a status output.
-  It can be ignored for this proposal.
-* `CredentialIssuer.status.strategies[*].frontend.impersonationProxyInfo.certificateAuthorityData` is a status output.
-  It can be ignored for this proposal.
+#### `WebhookAuthenticator`
+The [`validateTLSBundle`](https://github.com/vmware-tanzu/pinniped/blob/main/internal/controller/authenticator/webhookcachefiller/webhookcachefiller.go#L266) function will be updated to read and validate certificate authority data sourced using the newly added `certificateAuthorityDataSource` field.
+This function will continue to set the `TLSConfigurationValid` condition based on the validity.
 
-## Implementation
+#### `JWTAuthenticator`
+The [`validateTLS`](https://github.com/vmware-tanzu/pinniped/blob/main/internal/controller/authenticator/jwtcachefiller/jwtcachefiller.go#L248) function will be updated to read and validate certificate authority data sourced using the newly added `certificateAuthorityDataSource` field.
+This function will continue to set the `TLSConfigurationValid` condition based on the validity.
+
+
+For all the above custom resources, spanning both supervisor and concierge components, the informers that trigger their reconciliation will now watch for secrets, of type `Opaque` and `kubernetes.io/tls`, and configmaps to ensure any changes to the certificate authority data are hot loaded into respective custom resources.
+
+## Security Implications
+
+1. Pinniped will be able to read all of the data in the secrets and configmaps used to configure TLS CA bundles.
+2. For secrets of type tls, this means that Pinniped will be able to read both public and private keys in the tls secret.
+3. Pinniped will also be able to read all of the secrets and configmaps in the same namespace where Pinniped is running.
+
+## Validations and Status
```

Comment:
We should also validate that the `tls.certificateAuthorityDataSource.kind` is either `ConfigMap` or `Secret`. This could be a kubebuilder enum validation declared on the struct.